lib/srv/db/cloud/watchers/azure.go

```diff
@@ -0,0 +1,186 @@
+func newAzureFetcher(config azureFetcherConfig) (*azureFetcher, error) {
+	if err := config.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	fetcher := &azureFetcher{
+		cfg: config,
+		log: logrus.WithFields(logrus.Fields{
+			trace.Component: "watch:azure",
+			"labels":        config.Labels,
+			"regions":       config.Regions,
+			"group":         config.ResourceGroup,
+			"subscription":  config.Subscription,
+		}),
+	}
+	return fetcher, nil
+}
+
+// azureFetcherConfig is the Azure database servers fetcher configuration.
+type azureFetcherConfig struct {
+	// Client is the Azure API client.
+	Client azure.DBServersClient
+	// Subscription is the Azure subscription being fetched from by the fetcher.
+	Subscription string
+	// ResourceGroup is a selector to match cloud resource group.
+	ResourceGroup string
+	// Labels is a selector to match cloud databases.
+	Labels types.Labels
+	// Regions is the Azure regions selectors to match cloud databases.
+	Regions []string
+	// regionMatches returns whether a given region matches the configured Regions selector
+	regionMatches func(region string) bool
+}
+
+// CheckAndSetDefaults validates the config and sets defaults.
+func (c *azureFetcherConfig) CheckAndSetDefaults() error {
+	if c.Client == nil {
+		return trace.BadParameter("missing parameter Client")
+	}
+	if len(c.Subscription) == 0 {
+		return trace.BadParameter("missing parameter Subscription")
+	}
+	if len(c.ResourceGroup) == 0 {
+		return trace.BadParameter("missing parameter ResourceGroup")
+	}
+	if len(c.Labels) == 0 {
+		return trace.BadParameter("missing parameter Labels")
+	}
+	if len(c.Regions) == 0 {
+		return trace.BadParameter("missing parameter Regions")
+	}
+	if apiutils.SliceContainsStr(c.Regions, types.Wildcard) {
+		// wildcard matches all regions
+		c.regionMatches = func(_ string) bool { return true }
+	} else {
+		regionSet := utils.StringsSet(c.Regions)
+		c.regionMatches = func(region string) bool {
+			_, ok := regionSet[region]
+			return ok
+		}
+	}
+	return nil
+}
+
+// azureFetcher retrieves Azure DB single-server databases.
+type azureFetcher struct {
+	cfg azureFetcherConfig
+	log logrus.FieldLogger
+}
+
+// Get returns Azure DB servers matching the watcher's selectors.
+func (f *azureFetcher) Get(ctx context.Context) (types.Databases, error) {
+	databases, err := f.getDatabases(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return filterDatabasesByLabels(databases, f.cfg.Labels, f.log), nil
+}
+
+// getDatabases returns a list of database resources representing Azure database servers.
+func (f *azureFetcher) getDatabases(ctx context.Context) (types.Databases, error) {
+	var servers []*azure.DBServer
+	var err error
+	if f.cfg.ResourceGroup == types.Wildcard {
+		servers, err = f.cfg.Client.ListAll(ctx, common.MaxPages)
+	} else {
+		servers, err = f.cfg.Client.ListWithinGroup(ctx, f.cfg.ResourceGroup, common.MaxPages)
+	}
```

Comment:
nit: Can we move this into separate function  so the additional vars won't be needed
```suggestion
	servers, err := s.getDBServers(ctx, ...)
```

Reply:
Sure